### PR TITLE
Integrating agent installer with e2e

### PR DESCRIPTION
**Pull request by jhou1**
Integrate E2E with Agent installer workflow.

/assign @liangxia 

/hold until the test finishes

### CODE REVIEW

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 1]
```diff
@@ -2902,6 +2902,22 @@ tests:
     test:
     - chain: openshift-e2e-test-qe
     workflow: cucushift-installer-rehearse-vsphere-upi-disconnected-secureboot
+- as: baremetal-sno-agent-ipv4-static-connected-p1-f14
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2905]: Would you mind to sort the job names ?

Comment (jhou1) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml]: Done.

**Comment by jhou1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-baremetal-sno-agent-ipv4-static-connected-p1-f14

**Comment by jhou1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f14

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 2]
```diff
@@ -2095,6 +2095,22 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-baremetalds-ipi-ovn-ipv4
+- as: baremetal-sno-agent-ipv4-static-connected-p1-f14
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2098]: Missing `E2E_RUN_TAGS` for this job

Comment (jhou1) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2098]: Thanks, I should also use baremetal-upi tag and specify arm64/amd64,  will fix them together in the next commit.

**Comment by jhou1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f14

**Comment by jhou1** [comment]
Ready for review.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 3]
```diff
@@ -2095,6 +2095,24 @@ tests:
...
+  cluster: build05
+  cron: 30 21 * * 2
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2100]: Runs on every Tuesday, does not match the frequency f14.

**Comment by jhou1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f14

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml (hunk 3 above)
Comment (jhou1) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2100]: Fixed, thanks for review

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__arm64-nightly.yaml [hunk 4]
```diff
@@ -1525,6 +1525,46 @@ tests:
...
+      E2E_RUN_TAGS: '@baremetal-upi and @arm64 and @network-ovnkubernetes and not
+        @fips'
+      TAG_VERSION: '@4.13'
+      architecture: arm64
+      masters: "3"
+      workers: "2"
+    test:
+    - chain: openshift-e2e-test-qe
+    workflow: baremetal-lab-upi
+>>>>>>> 8f28c1dbac0 (integrating agent installer with e2e)
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__arm64-nightly.yaml:1567]: Unexpected line ?

Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__arm64-nightly.yaml:1567]: Unexpected line ?

**Comment by jhou1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by aleskandro** [comment]
Hi @jhou1 can you cherry-pick the commit ab231aa41b21cbbb0414e5972ad3f0f3a8f0be1e in your branch? It's not a definitive solution but helps maintaining the status of virtual media caching in the Ampere servers coherent.

**Comment by jhou1** [comment]
/pj-rehearse ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by aleskandro** [comment]
> No IP Address has been set for the external API VIP, failing


This is a quota issue in the rdu2 side, also other jobs and infras are affected.

**Comment by jhou1** [comment]
Thanks @aleskandro @liangxia !

I'd wait for the infra issue to be resolved and then rerun the tests before this gets merged.

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by jhou1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-baremetal-sno-agent-ipv4-static-connected-p1-f7

**Comment by aleskandro** [comment]
The last testrun seems using a cached version of the previous test. Ampere's redfish support is a mess.

**Comment by jhou1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-baremetal-sno-agent-ipv4-static-connected-p1-f7

File: ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh [hunk 5]
```diff
@@ -293,14 +319,16 @@ for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
...
+   "${iso_path}" "${transfer_protocol_type}" &
 done
 
+wait
+if [ -f /tmp/virtual_media_mount_failed ]; then
+  echo "Failed to mount the ISO image in one or more hosts"
+  exit 1
+fi
+
 for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh:332]: `# shellcheck disable=SC2154` before the for loop to fix the failing shell check test

**Comment by jhou1** [comment]
This PR is now ready to merge.

/pj-rehearse ack

**Pull request merged**